### PR TITLE
[Consensus] improve logs and panic messages

### PR DESCRIPTION
**Pull request by mwtian**
## Description 

Improve panic message when local timestamp < accepted block timestamp.

Add more details to logs for block creation and processing.

Remove logic no longer necessary after using DagState for proposing and keeping track of last proposed ancestors.

Ensure timestamp_utc_ms() returns monotonic values. Ensure after Core recover, local time > max ancestor timestamp.

## Test plan 

CI
Simulation

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: consensus/core/src/core.rs [hunk 1]
```diff
@@ -424,43 +435,9 @@ impl Core {
...
-        assert!(!to_propose.is_empty());
-
-        // always include our last proposed block in front of the vector and make sure that we do not
-        // double insert.
-        let mut result = vec![self.last_proposed_block.reference()];
-        for (authority_index, block_ref) in to_propose {
-            if authority_index != self.context.own_index {
-                result.push(block_ref);
-            }
-        }
```
Comment (akichidis) [inline consensus/core/src/core.rs:461]: nice clean up on the DAG compression, indeed there isn't much point keeping this

File: consensus/core/src/core.rs [hunk 2]
```diff
@@ -274,8 +269,24 @@ impl Core {
...
+        // Determine the ancestors to be included in proposal
+        let ancestors = self.ancestors_to_propose(clock_round);
+        self.context
+            .metrics
+            .node_metrics
+            .block_ancestors_count
+            .inc_by(ancestors.len() as u64);
+
+        // Ensure ancestor timestamps are not more advanced than the current time.
+        // Also catch the issue if system's clock go backwards (ex due to ntp corrections)
```
Comment (akichidis) [inline consensus/core/src/core.rs:281]: regarding the ntp corrections, I explicitly had that looked up and it looks like ntp will never attempt setting the clock backwards . But I assume things can still happen from some manual time change

**Comment by akichidis** [comment]
> If the new block timestamp is determined before ancestor blocks are read from store, it is possible that an ancestor block can include a later timestamp because the propose logic can be paused while new blocks are accepted. Instead, determine the new block timestamp after reading ancestor blocks, then check the invariant.

I might be missing something, but since both the proposal and block acceptance always go via Core, and run via a single thread loop, how is it possible to accept new blocks while the new block proposal has not finished yet? I don't believe we do have any path that inserts blocks to DagState directly.

**Comment by mwtian** [comment]
It is indeed strange. Commit sync does not get activated in the simulation. There is no evidence of clock getting adjusted. Let's see tomorrow's simulation result as well.

File: consensus/core/src/core.rs [hunk 3]
```diff
@@ -335,7 +358,7 @@ impl Core {
             .into_iter()
             .for_each(TransactionGuard::acknowledge);
 
-        info!("Created block {}", verified_block);
+        info!("Created block {:?}", verified_block);
```
Comment (akichidis) [inline consensus/core/src/core.rs:361]: I am thinking whether that should be a debug as well

File: consensus/core/src/core.rs [hunk 4]
```diff
@@ -274,8 +281,24 @@ impl Core {
...
         // Probably proposing for all the intermediate rounds might not make much sense.
 
-        // Consume the ancestors to be included in proposal
-        let ancestors = self.ancestors_to_propose(clock_round, now);
+        // Determine the ancestors to be included in proposal
+        let ancestors = self.ancestors_to_propose(clock_round);
+        self.context
+            .metrics
+            .node_metrics
+            .block_ancestors_count
```
Comment (akichidis) [inline consensus/core/src/core.rs:289]: Depending on the level of info alternatively could use a histogram?

**Review by akichidis — APPROVED** [review]

File: consensus/core/src/core.rs (hunk 4 above)
Comment (mwtian) [reply consensus/core/src/core.rs]: Sounds good.

**Review by arun-koshy — APPROVED** [review]

File: consensus/core/src/core.rs [hunk 5]
```diff
@@ -253,16 +261,15 @@ impl Core {
...
-
         // Create a new block either because we want to "forcefully" propose a block due to a leader timeout,
         // or because we are actually ready to produce the block (leader exists and min delay has passed).
         if !force {
             if !self.last_quorum_leaders_exist() {
                 return None;
             }
-            if Duration::from_millis(now.saturating_sub(self.last_proposed_timestamp_ms()))
-                < self.context.parameters.min_round_delay
+            if Duration::from_millis(
```
[the remark this answers is not in the document]
Comment (mwtian) [reply consensus/core/src/core.rs:270]: The 2nd if statement is a bit long so keeping them separate for now. We can merge them if needed.

**Review by arun-koshy — APPROVED** [review]

**Pull request merged**